db/migrate/20210215161632_create_user_events.rb

```diff
@@ -0,0 +1,11 @@
+class CreateUserEvents < ActiveRecord::Migration[6.0]
+  def change
+    create_table :user_events do |t|
+      t.integer :event_type
+      t.json :event_data
+
+      t.timestamps
+    end
+    add_index :user_events, :event_type
```

Comment:
I guess I may not have needed to index this but I did since it will be queried most often with the bonus question seeking daily event activity.